### PR TITLE
feat(rightmenu): Add Datasets to + Menu and Hide Databases when one has been connected

### PR DESCRIPTION
**Pull request by Antonio-RiveroMartnez**
### SUMMARY
We are adding some extra logic to RightMenu so it checks for what databases the user has connected before displaying options in the dropdown.

```
Plus menu logic:

Connect database: shows for users who can create databases, until their first non examples db is connected
Create dataset: shows for users who can create a dataset as soon as at least one non examples db is connected
[no change] Connect Google Sheet: shows for users who can create databases, always
[no change] Upload CSV, columnar, Excel: shows for users who can create databases all the time (tooltip when no csv uploads enabled, active when csv uploads enabled). For users who can upload but can't connect db, only shows once uploads are enabled on at least one db.
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
error

After:
test
test

### TESTING INSTRUCTIONS
1. Login into the app
2. Connect a database that is not example and check the logic form above
3. Remove all connected databases and check the logic from above
4. Clicking on the connect database menu should open the Connect Database modal
5. Clicking on the create dataset menu should open the Create Dataset modal

### ADDITIONAL INFORMATION

- [X] Has associated issue: https://github.com/apache/superset/issues/21569
- [ ] Required feature flags:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/components/RightMenu.test.tsx [hunk 1]
```diff
@@ -0,0 +1,367 @@
+/**
```
Comment (eschutho) [inline superset-frontend/src/views/components/RightMenu.test.tsx:1]: awesome a new test!!

**Review by eschutho — APPROVED** [review]
Looks great, and thanks for the speedy improvement on the dropdown menu, too!

**Comment by jinghua-qa** [comment]
LGTM!

**Comment by jinghua-qa** [comment]
Test the alpha and gamma role in local env and verified alpha user can see create dataset in + menu when 1 non-example db is added.

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Pull request merged**